pkg/controller/common.go

```diff
@@ -531,20 +549,33 @@ func (c *SharedState) addPod(pod *v1.Pod, updateOption updateWithLock) {
 		allClaims = append(allClaims, key.(string))
 	}
 	c.podToClaimsMap.Store(podKey, allClaims)
+	return nil
 }
 
-func (c *SharedState) deletePod(podKey string) {
+func (c *SharedState) deletePod(ctx context.Context, podKey string) error {
 	value, exists := c.podLocks.LoadAndDelete(podKey)
 	if !exists {
-		return
+		return nil
 	}
 	podLock := value.(*sync.Mutex)
 	podLock.Lock()
 	defer podLock.Unlock()
 
+	value, exists = c.podToInlineMap.LoadAndDelete(podKey)
+	if exists {
+		inlines := value.([]string)
+
+		for _, inline := range inlines {
+			if err := c.azClient.DiskV1alpha2().AzVolumes(c.objectNamespace).Delete(ctx, inline, metav1.DeleteOptions{}); err != nil {
```

Comment:
We probably should be safe and return an error only if the error something other than "not found".